go/libraries/doltcore/schema/typeinfo/point.go

```diff
@@ -45,13 +46,17 @@ func (ti *pointType) ConvertNomsValueToValue(v types.Value) (interface{}, error)
 	if _, ok := v.(types.Null); ok || v == nil {
 		return nil, nil
 	}
-
+	var err = fmt.Errorf(`"%v" cannot convert NomsKind "%v" to a value`, ti.String(), v.Kind())
 	// Expect a types.Point, return a sql.Point
 	if val, ok := v.(types.Point); ok {
-		return ConvertTypesPointToSQLPoint(val), nil
+		sqlVal := ConvertTypesPointToSQLPoint(val)
+		err = ti.sqlPointType.MatchSRID(sqlVal)
```

Comment:
I'm wondering if we actually **need** to validate the SRID here... we definitely need to validate any new data coming into the database from the caller (of course) and we need to validate any schema changes to make sure they don't invalidate the SRID matching constraint, but I'm wondering if we can be more lenient when we're loading data into and out of the storage layer interface. The impact of this validation failing is that customer won't be able to load their (already entered into the system) data any more, which could be worse than it being loaded and having inconsistent SRIDs that don't fit the constraint of the column. 

Are we testing this constraint in enough other places where the data is coming into the system so that we don't need to test this again as the data goes into the storage layer? I'm not confident yet, but wanted to bring this up to consider.